external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSource.scala

```diff
@@ -511,12 +537,20 @@ private[kafka010] object KafkaSource {
   ))
 
   sealed trait ConsumerStrategy {
-    def createConsumer(): Consumer[Array[Byte], Array[Byte]]
+    def createConsumer(kafkaParams: ju.Map[String, Object]): Consumer[Array[Byte], Array[Byte]]
+
+    def newKafkaParams(
+        kafkaParams: ju.Map[String, Object],
+        groupId: String): ju.Map[String, Object] = {
+      val newKafkaParams = new ju.HashMap[String, Object](kafkaParams)
```

Comment:
Nit: please dont reuse the name of the function as a variable. its confusing while reading code.